src/Result/category_result.py

```
@@ -0,0 +1,99 @@
+import time
+
+from src.Result.task_result import TaskResult
+from src.TaskConfiguration.task_category import TaskCategory
+
+
+class CategoryResult:
+
+    DISABLED_PREFIX = "#"
+
+    def __init__(self, category: TaskCategory):
+        self.category = category
+        self.enabled = category.enabled
+        self.task_results = []
+        self.category_results = []
+        self._started = time.time()
+        self._ended = None
+
+    def add_task_result(self, task_result: TaskResult):
+        self._raise_if_ended()
+        self.task_results.append(task_result)
+
+    def add_category_result(self, category_result):
+        self._raise_if_ended()
+        self.category_results.append(category_result)
+
+    def finalize(self):
+        self._raise_if_ended()
+        self._ended = time.time()
+
+        return self
+
+    def has_results(self):
+        self._raise_if_not_ended()
```

I'm not pretty sure we should raise an exception here.
Someone could use this function to check whether results are present (for whatever reason) although results are not finalized.